src/applications/personalization/dashboard-2/components/claims-and-appeals/ClaimsAndAppeals.unit.spec.jsx

```diff
@@ -0,0 +1,1088 @@
+          expect(view.queryByRole('heading', { name: /^claims & appeals$/i }))
+            .to.not.exist;
+        });
+      },
+    );
+    context(
+      'when user has both the `evss-claims` and `appeals-status` services',
+      () => {
+        beforeEach(() => {
+          mockFetch();
+          initialState = {
+            user: {
+              profile: {
+                services: ['evss-claims', 'appeals-status'],
+              },
+            },
+          };
+          view = renderInReduxProvider(<ClaimsAndAppeals />, {
+            initialState,
+            reducers,
+          });
+        });
+        afterEach(() => {
+          resetFetch();
+        });
+        it('should attempt to get appeals and claims data', async () => {
+          // Because fetch is called as part of an async Redux thunk, we need to
+          // wait here before confirming that fetch was called or not called.
+          await wait(1);
+          const fetchCalls = global.fetch.getCalls();
+          expect(
+            fetchCalls.some(call => {
+              return call.args[0].includes('v0/appeals');
+            }),
+          ).to.be.true;
+          expect(
+            fetchCalls.some(call => {
+              return call.args[0].includes('v0/evss_claims_async');
+            }),
+          ).to.be.true;
+        });
+        it('should render a loading spinner but no section headline while loading data', () => {
+          expect(view.queryByRole('progressbar', { label: /loading/i })).to
+            .exist;
+          expect(view.queryByRole('heading', { name: /^claims & appeals$/i }))
+            .to.not.exist;
+        });
+      },
+    );
+  });
+
+  describe('render logic', () => {
+    context('when the user has no claims or appeals on file', () => {
+      beforeEach(() => {
+        initialState = {
+          user: claimsAppealsUser(),
+          disability: {
+            status: {
+              claimsV2: {
+                appealsLoading: false,
+                claimsLoading: false,
+                appeals: [],
+                claims: [],
+              },
+            },
+          },
+        };
+        view = renderInReduxProvider(<ClaimsAndAppeals dataLoadingDisabled />, {
+          initialState,
+          reducers,
+        });
+      });
+      it('does not render anything, even a headline', () => {
+        expect(view.queryByRole('progressbar')).to.not.exist;
+        expect(view.queryByRole('heading', { name: /^claims & appeals$/i })).to
+          .not.exist;
+      });
+    });
+    context(
+      'when the user has 3 claims that were all updated in the past 30 days',
+      () => {
+        beforeEach(() => {
+          initialState = {
+            user: claimsAppealsUser(),
+            disability: {
+              status: {
+                claimsV2: {
+                  appealsLoading: false,
+                  claimsLoading: false,
+                  appeals: [],
+                  claims: [
```

Comment:
I want to clean up this test file a bit, adding helpers to generate the claims and appeals data we want for each test to make things easier to read. Feels like half this file size is just the mock claims and appeals data 😬